ci/install-musl.sh

```diff
@@ -64,13 +64,62 @@ esac
 cd ..
 rm -rf "$musl"
 
-# Download, configure, build, and install musl-sanitized kernel headers:
-kernel_header_ver="4.19.88"
-curl --retry 5 -L \
-    "https://github.com/sabotage-linux/kernel-headers/archive/v${kernel_header_ver}.tar.gz" |
-    tar xzf -
+# Download, configure, build, and install musl-sanitized kernel headers.
+
+# Alpine follows stable kernel releases, 3.20 uses Linux 6.6 headers.
+alpine_version=3.20
+alpine_git=https://gitlab.alpinelinux.org/alpine/aports
+
+# This routine piggybacks on: https://git.alpinelinux.org/aports/tree/main/linux-headers?h=3.20-stable
+git clone -n --depth=1 --filter=tree:0 -b "${alpine_version}-stable" "$alpine_git"
 (
-    cd "kernel-headers-${kernel_header_ver}"
-    make ARCH="${kernel_arch}" prefix="/musl-${musl_arch}" install -j4
+   cd aports
+   git sparse-checkout set --no-cone main/linux-headers
+   git checkout
+
+   cd main/linux-headers
+   cp APKBUILD APKBUILD.vars
+   {
+   echo "printf \"\$source\" > alpine-source"
+   # shellcheck disable=SC2028
+   echo "printf \"\$_kernver\n\" > alpine-kernver"
+   # shellcheck disable=SC2028
+   echo "printf \"\$pkgver\n\" > alpine-pkgver"
+   # shellcheck disable=SC2028
+   echo "printf \"\$sha512sums\n\" > alpine-sha512sums"
+   } >> APKBUILD.vars
```

Comment:
Could this just use heredoc rather than the nested quoting? Which shouldn't hit the shellcheck warnings